core/src/test/java/org/infinispan/eviction/impl/BaseEvictionFunctionalTest.java

```diff
@@ -60,17 +61,9 @@ public void testSimpleEvictionMaxEntries() throws Exception {
          cache.put("key-" + (i + 1), "value-" + (i + 1));
       }
       assertEquals(CACHE_SIZE, cache.size());
-      int expectedEvictions;
-      if (getEvictionStrategy() == EvictionStrategy.LIRS) {
-         // Eviction count will be Size + (Size * .05) rounded up (since the first elements will be in resident blocks
-         // so they won't cause evictions to occur
-         expectedEvictions = (int) Math.ceil(CACHE_SIZE + (CACHE_SIZE * .05));
-      } else {
-         // Otherwise is LRU and that will evict on each write
-         expectedEvictions = CACHE_SIZE * 2;
-      }
-      assertEquals("eviction events count should be same with case size: " + evictionListener.getEvictedEvents(),
-            expectedEvictions, evictionListener.getEvictedEvents().size());
+      // We don't know for sure how many will be evicted due to randomness, but we know they MUST evict
+      // at least a size worth since we are writing more than double
+      assertTrue(evictionListener.evictedEntries.size() > CACHE_SIZE);
```

Comment:
@ben-manes This test doesn't use weights at all, so that's not an issue.

The explanation was much simpler than what I was imagining: even if the test never accesses the keys after insertion, sometimes an older key will be retained instead of a newer one. Since the test inserts `2 * CACHE_SIZE` entries and then re-inserts the first `CACHE_SIZE` entries, the re-insert loop sometimes finds the entry already in the cache and doesn't have to evict anything. We could just change the test to always insert new keys in the cache, and the number of evictions would be guaranteed to be `2 * CACHE_SIZE`.